tests/lib/rules/no-new-func.js

```diff
@@ -21,7 +21,14 @@ const ruleTester = new RuleTester();
 ruleTester.run("no-new-func", rule, {
     valid: [
         "var a = new _function(\"b\", \"c\", \"return b+c\");",
-        "var a = _function(\"b\", \"c\", \"return b+c\");"
+        "var a = _function(\"b\", \"c\", \"return b+c\");",
+        {
+            code: "class Function {}; new Function()",
+            parserOptions: {
+                ecmaVersion: 2015
+            }
+        },
```

Comment:
It would be nice to have a test with the shadowing variable in an upper scope (but not the global scope), e.g.:

```js
var x = function Function() { Function(); }
```